platform/cosmos/stake.go

```diff
@@ -38,13 +42,40 @@ func (p *Platform) GetValidators() (blockatlas.ValidatorPage, error) {
 }
 
 func (p *Platform) GetDetails() blockatlas.StakingDetails {
-	//TODO: Find a way to have a dynamic
-	return blockatlas.StakingDetails{
-		Reward:        blockatlas.StakingReward{Annual: 11},
+	s := p.Coin().Symbol
+	if cached, err := p.Cache.GetCachedDetails(s); err == nil {
+		return *cached
+	}
+	nsd := blockatlas.StakingDetails{
+		Reward: blockatlas.StakingReward{
+			Annual: p.GetMaxAPR(),
+		},
 		MinimumAmount: blockatlas.Amount("0"),
 		LockTime:      1814400,
 		Type:          blockatlas.DelegationTypeDelegate,
 	}
+
+	if err := p.Cache.SaveCachedDetails(s, &nsd); err != nil {
```

Comment:
Is there a reason to save the entire object inside the database? Can we save only the APR to avoid save useless entries inside the database?

Reply:
Some other parameters can be dynamic in the future, that's why I decided to save the entire StakingDetails object

Reply:
great, make sense!

Reply:
I would only store what's needed

Reply:
I suggest to change Platform Storage interface to GetCachedMaxAPR() SetCachedMaxAPR() instead of GetCachedDetails() and SaveCachedDetails() 
@vikmeup